pkg/clusteragent/clusterchecks/dispatcher_isolate.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build clusterchecks
+
+package clusterchecks
+
+import "github.com/DataDog/datadog-agent/pkg/clusteragent/clusterchecks/types"
+
+func (d *dispatcher) isolateCheck(isolateCheckID string) types.IsolateResponse {
+	// Update stats prior to starting isolate to ensure all checks are accounted for
+	d.updateRunnersStats()
+	currentDistribution := d.currentDistribution()
+	isolateNode := currentDistribution.runnerForCheck(isolateCheckID)
+	if isolateNode == "" {
+		return types.IsolateResponse{
+			CheckID:   isolateCheckID,
+			CheckNode: "",
+			Result:    false,
+			Reason:    "Unable to find check",
+		}
+	}
+
+	proposedDistribution := newChecksDistribution(currentDistribution.runnerWorkers())
+
+	for _, checkID := range currentDistribution.checksSortedByWorkersNeeded() {
+		if checkID == isolateCheckID {
+			// Keep the check to be isolated on its current runner
+			continue
+		}
+
+		workersNeededForCheck := currentDistribution.workersNeededForCheck(checkID)
+		runnerForCheck := currentDistribution.runnerForCheck(checkID)
+
+		proposedDistribution.addToLeastBusy(
```

Comment:
I think this doesn't work correctly when there's only 1 runner in the cluster.

`addToLeastBusy` won't add the check because the only possible node is the isolated one, so `proposedDistribution` will be empty.
`applyDistribution` will not change anything, but the result will show that the check was isolated correctly.

It might be good to return an error when there's only 1 runner and multiple checks.